Config/app_state/api_factories.ts

```diff
@@ -0,0 +1,92 @@
+import produce from 'immer';
+import { SearchType, QueryType } from '../dev_env/types';
+
+const gutkaAPIFactory = ({ state, setState }) => {
+  const createGutka = (newName: string) => {
+    setState(
+      produce(draftState =>
+        draftState.gutkaNames.push(newName)));
+  }
+  const { gutkaNames, currentName, currentItems, isDataReady } = state;
+  return {
+    gutkaNames,
+    currentName,
+    currentItems,
+    isDataReady,
+
+    createGutka,
+  }
+}
+const globalApiFactory = ({ state, setState }) => {
+  const { currentName } = state;
+  const updateCurrentName = (newName: string) => {
+    setState(newName);
+  }
+  return { currentName, updateCurrentName }
+
+}
+const viewerApiFactory = ({ state, setState }) => {
+  const {
+    gurmukhiSize,
+    translSize,
+    translitSize,
+    displayEngTransl,
+    displayPunTansl,
+    displayTranslit
+  } = state;
+
+  const updateElementDisplay = (newVal: number, element: string) => {
+    const displaySetting = `display${element}`;
+    setState(prevState =>
+      ({
+        ...prevState,
+        [displaySetting]: newVal,
+      })
+    )
+  }
+  const updateFontSize = (newSize: number, element: number) => {
+    const fontSetting = `${element}Size`;
+    setState(prevState =>
+      ({
+        ...prevState,
+        [fontSetting]: newSize,
+      })
+
+    )
+  }
+  return {
+    gurmukhiSize,
+    translSize,
+    translitSize,
+    displayEngTransl,
+    displayPunTansl,
+    displayTranslit,
+
+    updateElementDisplay,
+    updateFontSize
+  }
+}
+const searchApiFactory = ({ state, setState }) => {
```

Comment:
just comment all these extra functions (basically everything thats not a react component)